app/routes/preview.py

```diff
@@ -0,0 +1,88 @@
+import json
+import os
+
+import boto3
+from fastapi import APIRouter, Request
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+
+from ..crud.sync_db.tile_cache_assets import get_dataset_tile_caches
+from ..settings.globals import GLOBALS
+
+router = APIRouter()
+
+template_dir = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), os.pardir, "templates"
+)
+templates = Jinja2Templates(directory=template_dir)
+
+
+@router.get(
+    "/{dataset}/{version}/{implementation}/preview", response_class=HTMLResponse
+)
+async def get_tile_caches(
+    *, request: Request, dataset: str, version: str, implementation
+):
+    """Map preview of availabe tile caches for a dataset."""
+
+    tile_caches = get_dataset_tile_caches(dataset, version, implementation)
+
+    sources = {
+        "carto-dark": {
+            "type": "raster",
+            "tiles": [
+                "https://a.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png",
+                "https://b.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png",
+                "https://c.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png",
+                "https://d.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png",
+            ],
+        },
+    }
+
+    layers = [
+        {
+            "id": "carto-dark-layer",
+            "type": "raster",
+            "source": "carto-dark",
+            "minzoom": 0,
+            "maxzoom": 22,
+        },
+    ]
+    for tile in tile_caches:
+        if tile["asset_type"] == "Static vector tile cache":
+            client = boto3.client("s3")
```

Comment:
I think you might want to use aioboto3 here instead since this is an async function. It's a library that just wraps boto3 in an async layer. Otherwise, this can be blocking for the application. Here's an example in the code where we use this for S3: https://github.com/wri/gfw-tile-cache/blob/ef3596fe4cdb92fc18b30706acb95f1975e2fb60/app/routes/raster_tiles.py#L187

Reply:
Good call, I'll switch to it.